src/react-integration/hooks/useResource.ts

```diff
@@ -55,17 +58,33 @@ function useManyResources<A extends ResourceArgs<any, any, any>[]>(
       Params extends Readonly<object>,
       Body extends Readonly<object> | void,
       S extends Schema
-    >([select, params]: ResourceArgs<S, Params, Body>) =>
+    >([fetchShape, params]: ResourceArgs<S, Params, Body>) =>
       // eslint-disable-next-line react-hooks/rules-of-hooks
-      useCache(select, params),
+      useCache(fetchShape, params),
+  );
+  const isExpiredList = resourceList.map(([fetchShape, params]) =>
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useIsExpired(fetchShape, params),
   );
   let promises = resourceList
-    .map(([select, params]) =>
+    .map(([fetchShape, params]) =>
       // eslint-disable-next-line react-hooks/rules-of-hooks
-      useRetrieve(select, params),
+      useRetrieve(fetchShape, params),
     )
     // only wait on promises without results
-    .filter((p, i) => p && !hasUsableData(resources[i], resourceList[i][0]));
+    .filter((p, i) => p && (!resources[i] || isExpiredList[i]));
+
+  // get the first error that is valid.
+  let error: null | Error = null;
+  for (let i = 0; i < resourceList.length; i++) {
+    const [fetchShape, params] = resourceList[i];
+    const resource = resources[i];
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const e = useError(fetchShape, params, resource);
+    if (e && !isExpiredList[i]) error = e;
```

Comment:
:nit: Do we need the `error` variable here, or could we just throw if we encounter an error and `! isExpiredList[i]`?